File: 68-text-justification/text-justification.cs

public class Solution {
    public IList<string> FullJustify(string[] words, int maxWidth) {
        var result = new List<string>();
        int index = 0;
        
        while (index < words.Length) {
            int totalChars = words[index].Length;
            int last = index + 1;
            while (last < words.Length) {
                if (totalChars + 1 + words[last].Length > maxWidth) break;
                totalChars += 1 + words[last].Length;
                last++;
            }
            
            StringBuilder line = new StringBuilder();
            int numOfWords = last - index;
            if (last == words.Length || numOfWords == 1) {
                for (int i = index; i < last; i++) {
                    line.Append(words[i]);
                    if (i < last - 1) line.Append(' ');
                }
                for (int i = line.Length; i < maxWidth; i++) {
                    line.Append(' ');
                }
            } else {
                int spaces = (maxWidth - totalChars) / (numOfWords - 1);
                int extraSpaces = (maxWidth - totalChars) % (numOfWords - 1);
                
                for (int i = index; i < last; i++) {
                    line.Append(words[i]);
                    if (i < last - 1) {
                        int spaceToApply = spaces + (i - index < extraSpaces ? 1 : 0);
                        for (int j = 0; j <= spaceToApply; j++) {
                            line.Append(' ');
                        }
                    }
                }
            }
            result.Add(line.ToString());
            index = last;
        }
        
        return result;
    }
}

File: 68-text-justification/text-justification.js
/**
 * @param {string[]} words
 * @param {number} maxWidth
 * @return {string[]}
 */
var fullJustify = function(words, maxWidth) {
    const result=[];
    let line=[],lineLength=0;
    for (let word of words){
        if (lineLength + line.length +word.length > maxWidth){
            for (let i=0; i<maxWidth -lineLength; i++){
                line[i % (line.length-1 || 1)] += ' ';
            }result.push(line.join(''));
            line=[];
            lineLength=0;
        }line.push(word);
        lineLength +=word.length;
    }result.push(line.join(' ')+' '.repeat(maxWidth-lineLength-line.length+1));
    return result;
};